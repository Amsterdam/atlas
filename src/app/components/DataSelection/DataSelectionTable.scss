.c-ds-table {
  $self: &;
  display: inline-table;
  width: 100%;
  border-collapse: collapse;

  &__head {
    display: table-header-group;
    white-space: nowrap;

    #{$self}__cell {
      border-bottom: 1px solid;
      border-top: 1px solid;
      font-weight: $bold-weight;
      height: 26px;
      vertical-align: middle;
      text-align: start;
    }
  }

  &__body {
    display: table-row-group;

    #{$self}__row {
      &--link {
        &:hover,
        &:focus {
          background-color: $secondary-gray10;
        }
      }

      &:last-child {
        #{$self}__cell {
          border-bottom: none;
        }
      }
    }
  }

  &__row {
    /*
     * Workaround for Chrome not supporting relative positions on table row elements. More information:
     * - https://bugs.chromium.org/p/chromium/issues/detail?id=417223
     * - https://stackoverflow.com/questions/8501727/table-row-wont-contain-elements-with-positionabsolute
     */
    transform: scale(1);
    display: table-row;
    text-decoration: none;
    position: relative;
  }

  &__cell {
    border-bottom: 1px solid $secondary-gray20;
    color: $primary-dark;
    display: table-cell;
    font-weight: $medium-weight;
    padding-bottom: $base-whitespace / 2;
    padding-left: $base-whitespace;
    padding-right: $base-whitespace * 2;
    padding-top: $base-whitespace / 2;
    white-space: nowrap;
  }
}
