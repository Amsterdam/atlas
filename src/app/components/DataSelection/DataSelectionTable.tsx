import { FunctionComponent } from 'react'
import styled from 'styled-components'
import { generatePath, Link } from 'react-router-dom'
import { getDetailPageData } from '../../../store/redux-first-router/actions'
import { routing } from '../../routes'
import DataSelectionFormatter from './DataSelectionFormatter/DataSelectionFormatter'
import { Data } from './types'

const TableRowLink = styled(Link)`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`

const DataSelectionTable: FunctionComponent<{ content: Data }> = ({ content }) =>
  content.body && content.body.length > 0 ? (
    <table className="c-ds-table" data-testid="dataSelectionTable">
      <thead className="c-ds-table__head">
        <tr className="c-ds-table__row c-ds-table__row--link">
          {content.head.map((field) => (
            <th key={field} className="c-ds-table__cell">
              {field}
            </th>
          ))}
        </tr>
      </thead>
      <tbody className="c-ds-table__body">
        {content.body.map((row) => (
          <tr
            key={row.id}
            className="c-ds-table__row c-ds-table__row--link qa-table-link"
            data-testid="dataSelectionTableRow"
          >
            {row.content.map((variables, j) => (
              <td key={variables[0].id} className="c-ds-table__cell">
                {/* @ts-ignore */}
                <DataSelectionFormatter
                  variables={variables}
                  formatter={content.formatters[j]}
                  template={content.templates[j]}
                />
              </td>
            ))}
            <TableRowLink
              to={generatePath(routing.dataDetail.path, getDetailPageData(row.detailEndpoint))}
            />
          </tr>
        ))}
      </tbody>
    </table>
  ) : null

export default DataSelectionTable
