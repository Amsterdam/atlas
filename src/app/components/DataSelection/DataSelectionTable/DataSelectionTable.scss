@import "../../../../shared/styles/config";

.c-ds-table {
  $self: &;
  display: inline-table;
  width: 100%;

  &__head {
    display: table-header-group;
    font-weight: $bold-weight;
    white-space: nowrap;

    #{$self}__cell {
      border-bottom: 1px solid;
      border-top: 1px solid;
      height: 26px;
      vertical-align: middle;
    }
  }

  &__body {
    display: table-row-group;

    #{$self}__row {

      &--link {
        &:hover,
        &:focus {
          background-color: $secondary-gray10;
        }
      }

      &:last-child {
        #{$self}__cell {
          border-bottom: none;
        }
      }
    }
  }

  &__row {
    display: table-row;
    text-decoration: none;
  }

  &__cell {
    border-bottom: 1px solid $secondary-gray20;
    color: $primary-dark;
    display: table-cell;
    font-weight: normal;
    padding-bottom: $base-whitespace / 2;
    padding-left: $base-whitespace;
    padding-right: $base-whitespace * 2;
    padding-top: $base-whitespace / 2;
    white-space: nowrap;
  }
}
