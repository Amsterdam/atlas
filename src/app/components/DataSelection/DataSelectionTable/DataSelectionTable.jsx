/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import Link from 'redux-first-router-link';
import DataSelectionFormatter from '../DataSelectionFormatter/DataSelectionFormatter';
import './DataSelectionTable.scss';
import { routing } from '../../../routes';
import { getDetailPageData } from '../../../../store/redux-first-router/actions';

const DataSelectionTable = ({ content }) => {
  const buildLink = (row) => {
    const { id, type, subtype } = getDetailPageData(row.detailEndpoint);
    return {
      type: routing.dataDetail.type,
      payload: {
        type,
        subtype,
        id: `id${id}`
      }
    };
  };

  return (
    (content.body && content.body.length > 0) && (
      <div className="c-ds-table">

        <div className="c-ds-table__head">
          <div className="c-ds-table__row">
            {content.head.map((field, i) => (
              <div key={i} className="c-ds-table__cell">
                {field}
              </div>
            ))}
          </div>
        </div>
        <div className="c-ds-table__body">
          {content.body.map((row) => (
            <Link
              key={row.detailEndpoint}
              className="c-ds-table__row"
              to={buildLink(row)}
            >
              {row.content.map((variables, i) => (
                <div
                  key={`${variables[0].value}_${variables[0].key}_${i}`}
                  className="c-ds-table__cell"
                >
                  <DataSelectionFormatter
                    variables={variables}
                    formatter={content.formatters[i]}
                    template={content.templates[i]}
                  />
                </div>
              ))}
            </Link>
          ))}
        </div>
      </div>)
  );
};

DataSelectionTable.propTypes = {
  content: PropTypes.shape({
    head: PropTypes.array,
    body: PropTypes.array
  }).isRequired
};

export default DataSelectionTable;
