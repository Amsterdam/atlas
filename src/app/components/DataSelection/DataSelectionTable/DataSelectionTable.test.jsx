import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import DataSelectionTable from './DataSelectionTable';

// Todo: DP-6235
describe('DataSelectionTable', () => {
  it('should render without failing', () => {
    const store = configureMockStore()({});
    const component = shallow(
      <DataSelectionTable
        content={{
          head: ['foo', 'bar'],
          body: [{ content: [], detailEndpoint: 'http://example.com' }]
        }}
      />,
      { context: { store } }
    ).dive();

    expect(component).toMatchSnapshot();
  });
});
