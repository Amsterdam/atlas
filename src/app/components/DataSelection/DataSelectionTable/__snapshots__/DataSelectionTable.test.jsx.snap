// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSelectionTable should render without failing 1`] = `
<table
  className="c-ds-table"
>
  <thead
    className="c-ds-table__head"
  >
    <tr
      className="c-ds-table__row c-ds-table__row--link"
    >
      <th
        className="c-ds-table__cell"
        key="0"
      >
        foo
      </th>
      <th
        className="c-ds-table__cell"
        key="1"
      >
        bar
      </th>
    </tr>
  </thead>
  <tbody
    className="c-ds-table__body"
  >
    <tr
      className="c-ds-table__row c-ds-table__row--link qa-table-link"
      key="0"
    >
      <Styled(mockConstructor) />
    </tr>
  </tbody>
</table>
`;
